check-lint

```diff
@@ -1,9 +1,102 @@
 #!/bin/sh -e
 
 TOOL_CMD=pylint
-PATH=$(basename $0)/utils:$PATH
-CONFIG_PATH="$(dirname $0)/default_configs/pylintrc"
-FILES=$(git ls-files '*.py')
+BASE_DIR=$(dirname "$0")
+CONFIG_FILE="$BASE_DIR/../avocado-static-checks.conf"
 
-echo "** Running $TOOL_CMD (using config from $CONFIG_PATH)..."
-python3 -m pylint --rcfile=$CONFIG_PATH $FILES
+# Default config path
+DEFAULT_CONFIG_PATH="$BASE_DIR/default_configs/pylintrc"
+
+# Collect all Python files in the repository
+ALL_FILES=$(git ls-files '*.py')
+
+# Initialize an array to store files checked with custom configs
+checked_files=()
+
+# Check if the avocado-static-checks.conf file exists
+if [ -f "$CONFIG_FILE" ]; then
+    echo "Found configuration file: $CONFIG_FILE"
+
+    # Initialize a flag to track if we're within the [lint] section
+    in_lint_section=false
+
+    # Read and process each line in the config file
+    while IFS=':' read -r directory config_path || [ -n "$directory" ]; do
+        # Skip empty lines
+        if [ -z "$directory" ] && [ -z "$config_path" ]; then
+            continue
+        fi
+
+        # Exit with an error if no directory is specified before the colon
+        if [ -z "$directory" ]; then
+            echo "Error: No directory specified before the colon in the config file."
+            exit 1
+        fi
+
+        # Handle section headers
+        if [[ "$directory" =~ ^\s*\[lint\]\s*$ ]]; then
+            in_lint_section=true
+            continue
+        elif [[ "$directory" =~ ^\s*\[.*\]\s*$ ]]; then
+            in_lint_section=false
+            continue
+        fi
+
+        # Process lines only within the [lint] section
+        if $in_lint_section; then
+            # Trim whitespace
+            CONFIG_PATH=$(echo "$config_path" | xargs)
+            DIRECTORY_PATH=$(echo "$directory" | xargs)
+
+            # Skip if either directory or config_path is empty
+            if [ -z "$DIRECTORY_PATH" ] || [ -z "$CONFIG_PATH" ]; then
+                echo "Warning: Skipping an invalid line in the config file."
```

Comment:
IMO the invalid line shouldn't be treated as warning, but as error. The reason for this is that this is probably caused by a user's mistake and if we will keep it as a warning, users might miss it.

Reply:
Thank you resolved it now